slash/loader.py

```diff
@@ -187,6 +187,6 @@ def _get_runnable_test_factory(self, thing):
 def _walk(p):
     if os.path.isfile(p):
         return [p]
-    return (os.path.join(dirname, filename)
-            for dirname, _, filenames in os.walk(p)
-            for filename in filenames)
+    return sorted([os.path.join(dirname, filename)
```

Comment:
this should be `sorted(os.path.join....` rather than `sorted([...` to prevent unnecessary list construction